### PR TITLE
Made main panel unscrollable

### LINKED ISSUE #87: Fix scroll behavior
The current version of the game allows two types of scroll.
+  within the paragraphs, which is intended for the user to read all the intro
+ within the whole page, which is not intended since it appears laggy and doest not bring anything (see video below)

https://github.com/roxannecvl/whokipedia/assets/66010389/c80075d1-2c67-4d9d-8829-864b2951fb34

Remove this useless second scroll by only enabling to scroll through paragraphs, picture and infobox. Search field should be kept on top, while the three others could be scrollable. Of course, main page should no longer be scrollable.

### LINKED ISSUE #95: Add remaining guesses
Following one of the peer review remarks, we should add the number of remaining guesses next to the current guess number. 
This could appear like : `Guess 3 out of 5`.

### PR DESCRIPTION
**Pull request by roxannecvl**
Closes #87
Closes #95

### CODE REVIEW

File: views/GameCenterView.vue [hunk 1]
```diff
@@ -1,7 +1,6 @@
 <script setup lang="ts">
 
 import type { ParagraphHint } from "~/model/Hint";
-import { getAutocompleteSuggestions } from "~/model/CelebrityList";
 import {blurHTML, getEncryptedString, removeNameOccurrences} from "~/utilities/Utils";
```
Comment (JEANPRBT) [inline views/GameCenterView.vue:4]:
```suggestion
import { blurHTML, getEncryptedString, removeNameOccurrences } from "~/utilities/Utils";
```

File: views/InfoboxView.vue [hunk 2]
```diff
@@ -28,56 +28,58 @@ const props = defineProps( {
...
+              'blur-none': blur === 0 || over,
+              'blur-sm': blur === 1,
+              'blur': blur === 2,
+              'blur-md': blur === 3,
+              'blur-lg': blur === 4,
+              'blur-xl': blur === 5,
+              'blur-2xl': blur === 6,
+              'blur-3xl': blur === 7
+            }"
+          />
```
Comment (JEANPRBT) [inline views/InfoboxView.vue:49]:
```suggestion
          <img :src="imageUrl"  alt="image" class="w-40 object-cover pb-2" :class="{
              'blur-none': blur === 0 || over,
              'blur-sm': blur === 1,
              'blur': blur === 2,
              'blur-md': blur === 3,
              'blur-lg': blur === 4,
              'blur-xl': blur === 5,
              'blur-2xl': blur === 6,
              'blur-3xl': blur === 7
            }"
          />
```
Again, you can save several lines.

File: app.vue [hunk 3]
```diff
@@ -38,9 +41,11 @@ const isDark = computed({
...
           <a href="/">
             <div class="flex">
-              <UIcon
-                  name="i-heroicons-magnifying-glass-16-solid"
-                  class="text-primary w-10 h-10 hidden sm:block"/>
+              <img
+                  :src="logoPath"
+                  alt="logo"
+                  class="w-9 h-9 mx-auto"
+              />
```
Comment (JEANPRBT) [inline app.vue:48]:
```suggestion
              <img :src="logoPath" alt="logo" class="w-9 h-9 mx-auto"/>
```

Keep this on one line.

File: views/InfoboxView.vue [hunk 4]
```diff
@@ -28,56 +28,58 @@ const props = defineProps( {
...
-      >
-        Learn more
-      </UButton>
-    </div>
-  </UCard>
+      <div v-if="over" style="display: flex; justify-content: center;">
+        <UButton
+            :to="buttonLink"
+            size="lg"
+        >
```
Comment (JEANPRBT) [inline views/InfoboxView.vue:79]:
```suggestion
        <UButton :to="buttonLink" size="lg">
```

File: views/SearchFieldView.vue [hunk 5]
```diff
@@ -0,0 +1,79 @@
+<script setup lang="ts">
+import {getAutocompleteSuggestions} from "~/model/CelebrityList";
```
Comment (JEANPRBT) [inline views/SearchFieldView.vue:2]:
```suggestion
import { getAutocompleteSuggestions } from "~/model/CelebrityList";
```

File: views/SearchFieldView.vue [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      v-model="selectedName"
+      :search="getAutocompleteSuggestions"
+      placeholder="Take a guess..."
+      option-attribute="name"
+      trailing
+      by="id"
+      :style="{ fontSize: '18px', padding: '10px', height: '40px',
+                      backgroundColor: redBackground ? redColor : '' }"
+      :class="{'tremble': tremble }"
+  />
```
Comment (JEANPRBT) [inline views/SearchFieldView.vue:59]:
```suggestion
  <UInputMenu v-if="!over" v-model="selectedName" :search="getAutocompleteSuggestions"
      placeholder="Take a guess..." option-attribute="name" trailing by="id" :style="{ 
         fontSize: '18px', 
         padding: '10px', 
         height: '40px',
         backgroundColor: redBackground ? redColor : '' 
     }" :class="{'tremble': tremble }"  />
```

File: views/SidebarView.vue [hunk 7]
```diff
@@ -49,34 +49,27 @@ function formatTime(seconds : number, over : boolean){
...
 
-  <UDivider
-      label="How to play"
-      :ui="{ label: 'text-xl font-bold' }"
-  />
+  <div>
+    <UDivider
+        label="How to play"
+        :ui="{ label: 'text-xl font-bold' }"
+    />
```
Comment (JEANPRBT) [inline views/SidebarView.vue:64]:
```suggestion
    <UDivider label="How to play" :ui="{ label: 'text-xl font-bold' }"/>
```

File: model/GameModel.ts [hunk 8]
```diff
@@ -54,8 +49,12 @@ export const useGameStore = defineStore('game', {
                     fetchIntro(this.name),
                 ])
                 this.images = images
+                this.totalGuesses += images.length - 1
                 this.infobox = infobox
+                this.totalGuesses += infobox.length - 1
```
Comment (JEANPRBT) [inline model/GameModel.ts:54]:
```suggestion
                this.totalGuesses += images.length + infobox.length - 2
```

Comment (JEANPRBT) [inline model/GameModel.ts:54]:
```suggestion
                this.infobox = infobox
                this.totalGuesses += infobox.length + totalGuesses - 2
```

File: model/GameModel.ts [hunk 9]
```diff
@@ -90,9 +95,18 @@ export const useGameStore = defineStore('game', {
...
                 }
             }
-        }
+        },
+        updateBlur(): number {
+            return this.images ?
+                this.images.filter((image: BlurHint) => image.revealed)
+                    .reduce((min, curr) => {
+                        return min.blur < curr.blur ? min : curr
+                    }).blur
```
Comment (JEANPRBT) [inline model/GameModel.ts:107]:
```suggestion
                    .reduce((min, curr) => min.blur < curr.blur ? min : curr ).blur
```

File: pages/daily-challenge.vue [hunk 10]
```diff
@@ -9,9 +9,30 @@ import SidebarPresenter from "~/presenters/SidebarPresenter.vue"
 // TODO: retrieve celebrity name from persistence
 const store: GameStore = useGameStore()
 store.init(getRandom(celebrities))
+const elapsedTime = ref(0)
 
 const isRulesOpen = ref(false)
```
Comment (JEANPRBT) [inline pages/daily-challenge.vue:14]:
```suggestion
// Refs
const elapsedTime = ref(0)
const isRulesOpen = ref(false)
```

File: views/GameCenterView.vue [hunk 11]
```diff
@@ -1,23 +1,24 @@
 <script setup lang="ts">
 
-import type { ParagraphHint } from "~/model/Hint";
-import { getAutocompleteSuggestions } from "~/model/CelebrityList";
-import {blurHTML, getEncryptedString, removeNameOccurrences} from "~/utilities/Utils";
+import type {InfoboxHint, ParagraphHint} from "~/model/Hint";
```
Comment (JEANPRBT) [inline views/GameCenterView.vue:3]:
```suggestion
import type { InfoboxHint, ParagraphHint } from "~/model/Hint";
```

**Review by JEANPRBT — APPROVED** [review]

**Review by MaelArrive — APPROVED** [review]

**Pull request merged**